engine/src/flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h

```diff
@@ -223,6 +224,25 @@ typedef NS_ENUM(NSInteger, FlutterAppExitResponse) {
  */
 @property(nonatomic, readonly) FlutterTextInputPlugin* textInputPlugin;
 
+@property(nonatomic, readonly) FlutterWindowController* windowController;
+
+/**
+ * Toggles multi-view support. Called by [FlutterWindowController] before
+ * creating a new window. This allows registering multiple view controllers
+ * with the engine.
```

Comment:
What happens in add-to-app scenarios where the native code immediately creates two `FlutterViewController`s at startup, before Dart isolate has launched? Will that work?

Reply:
The macOs embedder only support one view controller per engine. Right now only `FlutterWindowController` is blessed with the power to create multiple view controllers per engine. 

If we want to support that with public API I'd suggest doing that as separate PR. That would basically means having `_multiviewEnabled` on by default, but I'd prefer it in a small easily revertable PR in case we miss some edge case.